src/reducers/dataLoggerReducer.js

```diff
@@ -116,9 +157,23 @@ export default (state = initialState, { type, ...action }) => {
         case SET_DATA_LOGGER_STATE: {
             return { ...state, ...action };
         }
+        case MAX_BUFFER_SIZE: {
+            const { range } = state;
+            const { maxBufferSize } = action;
+            let { ranges, durationSeconds } = state;
+
+            ranges = getAdjustedRanges(maxBufferSize, ranges);
+            durationSeconds =
+                durationSeconds < range.max ? durationSeconds : range.max;
+
+            persistDuration(state.maxSampleFreq, durationSeconds);
+            persistMaxBufferSize(maxBufferSize);
+            return { ...state, ranges, maxBufferSize, durationSeconds };
```

Comment:
In general I'm not too fond of mutating, so perhaps change this to:

```suggestion
            const { ranges, durationSeconds } = state;

            updatesRanges= getAdjustedRanges(maxBufferSize, ranges);
            durationSeconds =
                durationSeconds < range.max ? durationSeconds : range.max;

            persistDuration(state.maxSampleFreq, durationSeconds);
            persistMaxBufferSize(maxBufferSize);
            return { ...state, ranges: updatesRanges, maxBufferSize, durationSeconds };
```